src/nl/hannahsten/texifyidea/completion/handlers/LatexReferenceInsertHandler.kt

```diff
@@ -23,21 +18,21 @@ class LatexReferenceInsertHandler(private val remote: Boolean = false, private v
         if (remote and TexifySettings.getInstance().automaticBibtexImport) {
             remoteBib ?: return
             // remoteBib may come from a file with CRLF line separators, which cannot be accepted into a psi file, so we need to fix that
-            val newBibEntry = LatexPsiHelper(context.project).createBibtexFromText(remoteBib.text.replace("\r\n", "\n")).findFirstChildOfType(BibtexEntry::class) ?: return
-
-            val bibsInFile = context.file.originalFile.bibtexIdsInFileSet()
-            // Add the bib item after the last item we found in the file set, and hope that that makes sense...
-            bibsInFile.lastOrNull()?.let {
-                it.parent?.addAfter(newBibEntry, it)
-            }
-
-            // If there are no bib items in the fileset yet, see if there is a(n empty) bib file we can add the bib entry to.
-            if (bibsInFile.isEmpty()) {
-                context.file.originalFile
-                    .referencedFileSet()
-                    .firstOrNull { it is BibtexFile }
-                    ?.add(newBibEntry)
-            }
+            // TODO: we should not iterate over all the bibtex entries, as this is very slow. Think of a better way to add the new bibtex entry.
```

Comment:
We indeed do not need all entries here, I think it would also be fine to find one and add the new one after it. Even more, we should look for bib files first as that is the most common case. That will still need the file set information though